mdoc/Mono.Documentation/Updater/Formatters/AttributeFormatters/CppWinRtAttributeFormatter.cs

```diff
@@ -0,0 +1,73 @@
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using Mono.Cecil;
+using Mono.Documentation.Util;
+
+namespace Mono.Documentation.Updater.Formatters
+{
+    class CppWinRtAttributeFormatter : AttributeFormatter
+    {
+        public override string PrefixBrackets { get; } = "/// [";
+        public override string SurfixBrackets { get; } = "]";
+        public override string Language => Consts.CppWinRt;
+
+        public override bool TryGetAttributeString(CustomAttribute attribute, out string rval, string prefix = null, bool withBrackets = true)
+        {
+            if (attribute == null)
+            {
+                if (string.IsNullOrEmpty(prefix))
+                {
+                    rval = null;
+                    return false;
+                }
+                rval = withBrackets ? PrefixBrackets + prefix + SurfixBrackets : prefix;
+                return true;
+            }
+
+            if (IsIgnoredAttribute(attribute))
+            {
+                rval = null;
+                return false;
+            }
+
+            TypeDefinition attrType = attribute.AttributeType as TypeDefinition;
+            if (attrType != null && !DocUtils.IsPublic(attrType)
+                || (FormatterManager.SlashdocFormatter.GetName(attribute.AttributeType) == null)
+                || Array.IndexOf(IgnorableAttributes, attribute.AttributeType.FullName) >= 0)
+            {
+                rval = null;
+                return false;
+            }
+
+            var fields = new List<string>();
+
+            for (int i = 0; i < attribute.ConstructorArguments.Count; ++i)
+            {
+                CustomAttributeArgument argument = attribute.ConstructorArguments[i];
+                string attributesValue = MakeAttributesValueString(argument.Value, argument.Type);
+                fields.Add(attributesValue.StartsWith("typeof(") ? attributesValue.Substring(7, attributesValue.Length - 8) : attributesValue);
```

Comment:
ok, so this is better ... but by overriding this particular method, you're really just having to copy/paste _the entire rest of the function_ here. It would make more sense to _not override_ `TryGetAttributeString`, but instead override `MakeAttributeValueString`.

You can make that a virtual protected method, and do something like:

```
var baseValue = base.MakeAttributesValueString(value, type);
return baseValue .StartsWith("typeof(") ? baseValue .Substring(7, baseValue .Length - 8) : baseValue;
```

This greatly reduces the amount of custom code in `CppWinRtAttributeFormatter`